catkin_ws/src/state_estimation/src/state_aggregator.py

```diff
@@ -7,65 +7,75 @@
 from tf2_ros import TransformBroadcaster
 
 from geometry_msgs.msg import Pose, Quaternion, Vector3, TransformStamped, Point
-from sensors import DepthSensor, IMU, DVL
-from std_msgs.msg import Float64, Bool
+from sensors import DepthSensor, IMU, IMUFrontCamera, DVL
+from std_msgs.msg import Float64, Int32
 from rosgraph_msgs.msg import Clock
 
 DEG_PER_RAD = 180 / np.pi
 
+def swap_main_sensor_message(axis_index, sensor_index, sensor_name):
+    global last_warning_message_time
+
+    if sensor_index != 0 and rospy.get_time() - last_warning_message_time[axis_index] > sensor_swap_warning_interval:
+            last_warning_message_time[axis_index] = rospy.get_time()
+            rospy.logwarn("Using {} for {}.".format(sensor_name, axis_names[axis_index]))
+
+
 def update_state(msg):
     if update_state_on_clock:
         global last_clock_msg_s
         global last_clock_msg_ns
         if last_clock_msg_s == msg.clock.secs and last_clock_msg_ns == msg.clock.nsecs: return
         last_clock_msg_s = msg.clock.secs
         last_clock_msg_ns = msg.clock.nsecs
-        
-    pub_dvl_sensor_status.publish(dvl.isActive())
-    pub_imu_sensor_status.publish(imu.isActive())
-    pub_depth_sensor_status.publish(depth_sensor.isActive())
-    x = None
-    y = None
-    z = None
+
+    position = [None, None, None]
```

Comment:
again, avoid repetition of code - use indexes for position instead of different variables